akd/tools/scrapers/pypaperbot.py

```diff
@@ -0,0 +1,330 @@
+
+            # Add optional parameters from config
+            if self.config.proxy:
+                cmd.extend(["--proxy", self.config.proxy])
+            if self.config.selenium_chrome_version:
+                cmd.extend(
+                    [
+                        "--selenium-chrome-version",
+                        str(self.config.selenium_chrome_version),
+                    ],
+                )
+
+            if self.debug:
+                logger.debug(f"Running PyPaperBot command: {' '.join(cmd)}")
+
+            process = await asyncio.create_subprocess_exec(
+                *cmd,
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+                cwd=out_dir,
+            )
+
+            try:
+                stdout, stderr = await asyncio.wait_for(
+                    process.communicate(),
+                    timeout=self.config.max_runtime_seconds,
+                )
+
+                if self.debug:
+                    logger.debug(f"PyPaperBot stdout: {stdout.decode()}")
+                    if stderr:
+                        logger.debug(f"PyPaperBot stderr: {stderr.decode()}")
+
+                return self._find_downloaded_pdf(out_dir)
+            except asyncio.TimeoutError:
+                process.kill()
+                if self.debug:
+                    logger.debug(f"PyPaperBot timeout for query: {query}")
+                return None
+        except Exception as e:
+            if self.debug:
+                logger.debug(f"PyPaperBot execution failed for query {query}: {e}")
+        return None
+
+    def _extract_doi_from_url(self, url: str) -> Optional[str]:
+        """Extract DOI from URL using simple pattern matching."""
+        import re
+
+        # Common DOI patterns
+        doi_patterns = [
+            r"10\.\d{4,}/[^\s\"<>#]+",  # Standard DOI format
+            r"/doi/(?:full/|pdf/|pdfdirect/)?(10\.[^/?#]+)",  # Wiley DOI extraction
+        ]
+
+        for pattern in doi_patterns:
+            match = re.search(pattern, url, re.IGNORECASE)
+            if match:
+                # Return the full DOI or the captured group
+                return match.group(1) if match.groups() else match.group(0)
+
+        return None
+
+    def _extract_query_from_url(self, url: str) -> Optional[str]:
+        """Extract a reasonable search query from URL by removing domain and using the rest."""
+        res = url
+        try:
+            parsed = urlparse(url)
+            # Use path + query as search term, removing leading slash
+            query_term = (parsed.path + parsed.query).lstrip("/")
+            # Clean up common file extensions
+            res = re.sub(r"\.(pdf|html?|aspx?)$", "", query_term, flags=re.IGNORECASE)
+        except Exception as e:
+            if self.debug:
+                logger.debug(f"Error extracting query from URL {url}: {e}")
+        return res
```

Comment:
I have modified this method because when simply url is passed to `--query` fallback process for pypaperbot, it will throw error.